app/src/main/java/org/stepik/android/adaptive/pdd/ui/fragment/CardsFragment.java

```diff
@@ -101,6 +101,24 @@ private void resolveDailyReward() {
             DailyRewardDialog.Companion.newInstance(progress).show(getChildFragmentManager(), DAILY_REWARD_DIALOG_TAG);
     }
 
+    private void resolveMigration() {
+        if (!SharedPreferenceMgr.getInstance().isMigrated()) {
+            onLoading();
+            compositeDisposable.add(MigrationHelper.migrate()
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .doOnError(this::onError)
```

Comment:
Please, classify errors of migration. The app should continue work if server with leaderboards will be shutted down